eng/pipelines/templates/scripts/assemble-dotnet-standalone-exe.ps1

```diff
@@ -0,0 +1,92 @@
+
+.PARAMETER Target 
+The targeted folder that should be built and assembled into a standalone executable.
+
+.PARAMETER Framework
+The targeted .NET framework. Defaults to "net6.0."
+
+#>
+param(
+   [Parameter(mandatory=$true)]
+   [string] $Rid,
+   [Parameter(mandatory=$true)]
+   [string] $Target,
+   [Parameter(mandatory=$true)]
+   [string] $ArtifactStagingDirectory,
+   [Parameter(mandatory=$false)]
+   [string] $Framework = "net6.0"
+)
+
+# resolves to <artifactfolder>/win-x64
+$destinationArtifactFolder = Join-Path $($ArtifactStagingDirectory) $($Rid)
+
+# resolves to <artifactfolder>/win-x64/test-proxy-standalone-win-x64 (.zip or .tar.gz will be added as appropriate for platform)
+$destinationPathSegment = Join-Path $destinationArtifactFolder "$((Split-Path -Leaf "$($Target)"))-standalone-$Rid"
+
+# resolves to tools/test-proxy/win-x64
+$outputPath = Join-Path $($Target) $($Rid)
+
+# ensure the destination artifact directory exists
+if (!(Test-Path $destinationArtifactFolder)){
+   New-Item -Force -Path $destinationArtifactFolder -ItemType directory
+}
+
+Write-Host "dotnet publish -f $($Framework) -c Release -r $($Rid) -p:PublishSingleFile=true --self-contained --output $outputPath $Target"
+dotnet publish -f $($Framework) -c Release -r $($Rid) -p:PublishSingleFile=true --self-contained --output $outputPath $Target
+
+if ($LASTEXITCODE -ne 0) {
+   Write-Error "dotnet publish failed with exit code $LASTEXITCODE."
+   exit $LASTEXITCODE
+}
+
+# produce a tar.gz only for linux
+if ("$($Rid)".Contains("linux")){
+   # tar on powershell in linux has some weirdness. For instance, this is a proper call to tar when we don't want to include the relative path to the target folder
+   # tar -cvzf -C tools/test-proxy/linux-arm64 blah.tar.gz tools/test-proxy/linux-arm64 
+   # however when we use this, we actually get an error. To avoid this, we simply CD into the target directory before tar-ing it.
+   Push-Location "$outputPath"
+   # The sum contents within this folder will be: `appSettings.json`, `test-proxy.pdb`, `test-proxy` (the binary), and a certificate.
+   # This statement grabs the first extensionless file within the produced binary folder, which will always be the binary we need to set the executable bit on.
+   $binaryFile = (Get-ChildItem -Path . | Where-Object { !([System.IO.Path]::HasExtension($_)) } | Select-Object -First 1).ToString().Replace("`\","/")
+   bash -c "chmod +x $binaryFile"
+   tar -cvzf "$($destinationPathSegment).tar.gz" .
+   Pop-Location
+}
+elseif("$($Rid)".Contains("osx")){
+   # need to codesign the binary with an entitlements file such that the signed and notarized binary will properly invoke on
+   # a mac system. However, the `codesign` command is only available on a MacOS agent. With that being the case, we simply special case
+   # this function here to ensure that the script does not fail outside of a MacOS agent.
+   if ($IsMacOS) {
```

Comment:
Missing some context, why do we not want it to fail outside of macos if the target platform is osx? Is it because it gets called anyway for the whole matrix, and we know that there are separate jobs handling the macos part?